### PR TITLE
[CARBONDATA-218]Using CSVInputFormat instead of spark-csv during dictionary geneartion

### PR DESCRIPTION
**Pull request by QiangCai**
1. Using CSVInputFormat instead of spark-csv during dictionary geneartion

2. Remove spark-csv dependency from whole project.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/429/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 1]
```diff
@@ -356,37 +363,49 @@ object GlobalDictionaryUtil {
...
-        }
-      })
-      .option("delimiter", {
-        if (StringUtils.isEmpty(carbonLoadModel.getCsvDelimiter)) {
-          "" + DEFAULT_SEPARATOR
-        } else {
-          carbonLoadModel.getCsvDelimiter
+      val hadoopConfiguration = new Configuration()
+      CommonUtil.configureCSVInputFormat(hadoopConfiguration, carbonLoadModel)
+      hadoopConfiguration.set(FileInputFormat.INPUT_DIR, carbonLoadModel.getFactFilePath)
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: use `FileInputFormat.addInput`

Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: FileInputFormat.addInputPath method need a Job type paramter.
In addition, this FactFilePath already consist of all file path, we can directly set input path, no need to separate path and add path again.

Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: ok

Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: ok

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**